.github/workflows/unit.yaml

```diff
@@ -35,3 +35,46 @@ jobs:
       if: ${{ !((github.event_name == 'repository_dispatch' && github.event.client_payload.slash_command.args.named.sha != '' && contains(github.event.client_payload.pull_request.head.sha, github.event.client_payload.slash_command.args.named.sha)) || inputs.skipRefCheck) }}
       with:
         script: core.setFailed('Ref is not latest')
+  go-cover:
+    needs: unit-test
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        ref: ${{ inputs.ref }}
+    - name: Set up Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: ~1.20.0
+    - name: Get gocover latest release asset url
+      id: get_release
+      uses: actions/github-script@v6
+      with:
+        result-encoding: string
+        script: |
+          const { data } = await github.rest.repos.getLatestRelease({
+            owner: 'Azure',
+            repo: 'gocover'
+          });
+          console.log(`Latest tag is ${data.tag_name}`);
+          console.log(`Assests count is ${data.assets.length}`);
+          const linuxAssetUrl = data.assets.filter(asset => asset.name.includes('linux_amd64'));
+          return linuxAssetUrl[0].browser_download_url;
+    - name: Download and run gocover
+      shell: bash
+      run: |
+        echo "download asset url from previous step is ${{ steps.get_release.outputs.result }}"
+        wget -q -O gocover.tar.gz ${{ steps.get_release.outputs.result }}
+        tar -xzf gocover.tar.gz
+        chmod +x gocover
+        echo $(pwd)
+        echo $(git branch)
+        echo $(git fetch --all)
+        ./gocover diff --cover-profile=coverage.out  --repository-path=./ --report-name=aks-app-routing-operator.report --outputdir=/tmp --compare-branch=origin/main  --coverage-baseline=80
+    - name: Upload coverage report
```

Comment:
It would be helpful if we could have a coverage report with coveralls like in the AgentBaker repo. https://github.com/Azure/AgentBaker/blob/d0a5f2a254a0209c927594484464c995c079555d/.github/workflows/check-coverage.yml#L38

It creates a nice comment on the PR breaking down where the coverage is for the PR. https://github.com/Azure/AgentBaker/pull/3471#issuecomment-1652335285

Reply:
https://github.com/Azure/aks-app-routing-operator/pull/74#discussion_r1277727332

Reply:
Done replaced with coverall.